src/api/client.ts

```diff
@@ -0,0 +1,197 @@
+  | "POST"
+  | "post"
+  | "DELETE"
+  | "delete"
+
+/**
+ * RequestOptions
+ * The set of options available for any http client request.
+ */
+export interface RequestOptions {
+  /**
+   * Disable the full screen loading interface during the request when true.
+   */
+  skipLoader?: boolean
+  /**
+   * Artificially delay a request by the time specified when set.
+   */
+  withDelay?: number
+}
+
+export const HTTP_ERROR = "HttpError"
+export const HTTP_CANCELLED_ERROR = "HttpCanceledError"
+
+/**
+ * HttpError
+ * An http client error when the request is rejected.
+ */
+export class HttpError<T = unknown> extends Error {
+  /**
+   * The http response status code.
+   */
+  status: number
+
+  /**
+   * The http response body.
+   */
+  response?: T
+
+  constructor(message?: string, status?: number, response?: T, name?: string) {
+    super(message || "")
+    this.name = name || HTTP_ERROR
+    this.status = status || 0
+    this.response = response
+  }
+}
+
+/**
+ * HttpPromise
+ * The successfully resolved interface of an http client request.
+ */
+export interface HttpPromise<T = any> extends Promise<T> {}
+
+/**
+ * TrailsPromise
+ * The successfully resolved interface of an http client request returned by @xy-planning-network/Trails.
+ */
+export interface TrailsPromise<T = any> extends Promise<TrailsResponse<T>> {}
+
+/**
+ * TrailsPromisePaged
+ * The successfully resolved interface of a paged http client request returned by @xy-planning-network/Trails.
+ */
+export interface TrailsPromisePaged<T = any>
+  extends Promise<TrailsResponsePaged<T>> {}
+
+/**
+ * TrailsResponse
+ * A convenience interface to represent the shape of a Trails delivered API response.
+ *
+ * Example Usage:
+ * interface User {
+ *     id: number
+ *     email: string
+ * }
+ *
+ * BaseAPI.get<TrailsResponse<User>>(`/user/${id}`)
+ *     .then(result => {
+ *         const id = result.data.id
+ *         const email = result.data.email
+ *         const user = { ...result.data }
+ *     })
+ */
+export interface TrailsResponse<T = any> {
+  data: T
+}
+
+/**
+ * TrailsResponsePaged
+ * A convenience interface to represent the shape of a paginated Trails delivered API response.
+ *
+ * Example Usage:
+ * interface User {
+ *     id: number
+ *     email: string
+ * }
+ *
+ * BaseAPI.get<TrailsResponsePaged<User>>(`/users`)
+ *     .then(result => {
+ *         const currentPage = result.data.page
+ *         const users = { ...result.data.items }
+ *
+ *         result.data.items.forEach(u => {
+ *             console.log(u.id)
+ *         })
+ *     })
+ */
+export interface TrailsResponsePaged<T = any> {
+  data: {
+    items: T[]
+    page: number
+    perPage: number
+    totalItems: number
+    totalPages: number
+  }
+}
+
+/**
+ * QueryParams
+ * Http GET request query params.
+ */
+export type QueryParams = Record<string, any>
+
+/**
+ * RequestPayload
+ * Http POST and PUT payloads.
+ */
+export type RequestPayload = Record<string, any> | FormData
```

Comment:
`RequestPayload` and `QueryParams` have been loosened to `Record<string, any>`.  It's been cumbersome to read things like `UserAPI.update({...user})` or `UserAPI.update(user as Record<string, unknown>)`.

Now nice and easy:
`UserAPI.update(user)`